collector/receiver/telemetryapireceiver/receiver_test.go

```diff
@@ -161,3 +170,301 @@ func TestCreatePlatformInitSpan(t *testing.T) {
+				require.Equal(t, tc.expectedLogRecords, scopeLog.LogRecords().Len())
+				if scopeLog.LogRecords().Len() > 0 {
+					logRecord := scopeLog.LogRecords().At(0)
+					attr, ok := logRecord.Attributes().Get("type")
+					require.True(t, ok)
+					require.Equal(t, tc.expectedType, attr.Str())
+					expectedTime, err := time.Parse(time.RFC3339, tc.expectedTimestamp)
+					require.NoError(t, err)
+					require.Equal(t, pcommon.NewTimestampFromTime(expectedTime), logRecord.Timestamp())
+					requestId, ok := logRecord.Attributes().Get(semconv.AttributeFaaSInvocationID)
+					require.Equal(t, tc.expectedContainsRequestId, ok)
+					if ok {
+						require.Equal(t, tc.expectedRequestId, requestId.Str())
+					}
+					require.Equal(t, tc.expectedSeverityText, logRecord.SeverityText())
+					require.Equal(t, tc.expectedSeverityNumber, logRecord.SeverityNumber())
+					require.Equal(t, tc.expectedBody, logRecord.Body().Str())
+				}
+			}
+		})
+	}
+}
+
+func TestSeverityTextToNumber(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		level  string
+		number plog.SeverityNumber
+	}{
+		{
+			level:  "TRACE",
+			number: plog.SeverityNumberTrace,
+		},
+		{
+			level:  "TRACE2",
+			number: plog.SeverityNumberTrace2,
+		},
+		{
+			level:  "TRACE3",
+			number: plog.SeverityNumberTrace3,
+		},
+		{
+			level:  "TRACE4",
+			number: plog.SeverityNumberTrace4,
+		},
+		{
+			level:  "DEBUG2",
+			number: plog.SeverityNumberDebug2,
+		},
+		{
+			level:  "DEBUG3",
+			number: plog.SeverityNumberDebug3,
+		},
+		{
+			level:  "DEBUG4",
+			number: plog.SeverityNumberDebug4,
+		},
+		{
+			level:  "INFO",
+			number: plog.SeverityNumberInfo,
+		},
+		{
+			level:  "INFO2",
+			number: plog.SeverityNumberInfo2,
+		},
+		{
+			level:  "INFO3",
+			number: plog.SeverityNumberInfo3,
+		},
+		{
+			level:  "INFO4",
+			number: plog.SeverityNumberInfo4,
+		},
+		{
+			level:  "WARN",
+			number: plog.SeverityNumberWarn,
+		},
+		{
+			level:  "WARN2",
+			number: plog.SeverityNumberWarn2,
+		},
+		{
+			level:  "WARN3",
+			number: plog.SeverityNumberWarn3,
+		},
+		{
+			level:  "WARN4",
+			number: plog.SeverityNumberWarn4,
+		},
+		{
+			level:  "ERROR",
+			number: plog.SeverityNumberError,
+		},
+		{
+			level:  "ERROR2",
+			number: plog.SeverityNumberError2,
+		},
+		{
+			level:  "ERROR3",
+			number: plog.SeverityNumberError3,
+		},
+		{
+			level:  "ERROR4",
+			number: plog.SeverityNumberError4,
+		},
+		{
+			level:  "FATAL",
+			number: plog.SeverityNumberFatal,
+		},
+		{
+			level:  "FATAL2",
+			number: plog.SeverityNumberFatal2,
+		},
+		{
+			level:  "FATAL3",
+			number: plog.SeverityNumberFatal3,
+		},
+		{
+			level:  "FATAL4",
+			number: plog.SeverityNumberFatal4,
+		},
+		{
+			level:  "CRITICAL",
+			number: plog.SeverityNumberFatal,
+		},
+		{
+			level:  "ALL",
+			number: plog.SeverityNumberTrace,
+		},
+		{
+			level:  "UNKNOWN",
+			number: plog.SeverityNumberUnspecified,
+		},
+	}
```

Comment:
I think this test simply specifies the severity map in another form. If you think it's beneficial I'd rather see the condensed format.

Reply:
I am not sure what do you mean by `Could move that map to top-level and create a separate map here as the oracle/golden.`. Can you help to elaborate by an simple example? Thanks!

Reply:
Done. Added the golden mapping table and out of boundary cases.